contracts/integrations/test/actors/staker.ts

```diff
@@ -44,6 +65,79 @@ export function StakerMixin<TBase extends Constructor>(Base: TBase): TBase & Con
+                yield assertion.executeAsync(amount, { from: this.actor.address });
+            }
+        }
+
+        private async *_validUnstake(
+            simulationEnvironment: SimulationEnvironment,
+        ): AsyncIterableIterator<AssertionResult> {
+            const { stakingWrapper } = this.actor.deployment.staking;
+            const { deployment, balanceStore, globalStake } = simulationEnvironment;
+            const assertion = validUnstakeAssertion(deployment, balanceStore, globalStake, this.stake);
+
+            while (true) {
+                await simulationEnvironment.balanceStore.updateErc20BalancesAsync();
+                const undelegatedStake = await stakingWrapper.getOwnerStakeByStatus.callAsync(
+                    this.actor.address,
+                    StakeStatus.Undelegated,
+                );
+                const withdrawableStake = BigNumber.min(
+                    undelegatedStake.currentEpochBalance,
+                    undelegatedStake.nextEpochBalance,
+                );
+                const amount = getRandomInteger(0, withdrawableStake);
+                yield assertion.executeAsync(amount, { from: this.actor.address });
+            }
+        }
+
+        private async *_validMoveStake(
+            simulationEnvironment: SimulationEnvironment,
+        ): AsyncIterableIterator<AssertionResult> {
+            const { deployment, globalStake } = simulationEnvironment;
+            const assertion = validMoveStakeAssertion(
+                deployment,
+                globalStake,
+                this.stake,
+                simulationEnvironment.stakingPools,
+            );
+
+            while (true) {
+                const fromPoolId = _.sample(Object.keys(_.omit(this.stake[StakeStatus.Delegated], ['total'])));
+                const fromStatus =
+                    fromPoolId === undefined
+                        ? StakeStatus.Undelegated
+                        : (_.sample([StakeStatus.Undelegated, StakeStatus.Delegated]) as StakeStatus);
+                const from = new StakeInfo(fromStatus, fromPoolId);
+
+                const toPoolId = _.sample(Object.keys(simulationEnvironment.stakingPools));
+                const toStatus =
+                    toPoolId === undefined
+                        ? StakeStatus.Undelegated
+                        : (_.sample([StakeStatus.Undelegated, StakeStatus.Delegated]) as StakeStatus);
+                const to = new StakeInfo(toStatus, toPoolId);
+
+                const moveableStake =
+                    from.status === StakeStatus.Undelegated
+                        ? this.stake[StakeStatus.Undelegated].nextEpochBalance
+                        : this.stake[StakeStatus.Delegated][from.poolId].nextEpochBalance;
+                const amount = getRandomInteger(0, moveableStake);
+
+                yield assertion.executeAsync(from, to, amount, { from: this.actor.address });
+            }
+        }
```

Comment:
These are really nice. As discussed offline, I think that we will want some generators that are not as "smart." Optional for this PR though.